central/imageintegration/datastore/datastore.go

```diff
@@ -16,17 +25,50 @@ var (
 // DataStore is the entry point for modifying Cluster data.
 //go:generate mockgen-wrapper
 type DataStore interface {
-	GetImageIntegration(ctx context.Context, id string) (*storage.ImageIntegration, bool, error)
-	GetImageIntegrations(ctx context.Context, integration *v1.GetImageIntegrationsRequest) ([]*storage.ImageIntegration, error)
+	GetImageIntegration(ctx context.Context, id string) (*genSt.ImageIntegration, bool, error)
+	GetImageIntegrations(ctx context.Context, integration *v1.GetImageIntegrationsRequest) ([]*genSt.ImageIntegration, error)
 
-	AddImageIntegration(ctx context.Context, integration *storage.ImageIntegration) (string, error)
-	UpdateImageIntegration(ctx context.Context, integration *storage.ImageIntegration) error
+	AddImageIntegration(ctx context.Context, integration *genSt.ImageIntegration) (string, error)
+	UpdateImageIntegration(ctx context.Context, integration *genSt.ImageIntegration) error
 	RemoveImageIntegration(ctx context.Context, id string) error
+	Search(ctx context.Context, q *v1.Query) ([]search.Result, error)
 }
 
 // New returns an instance of DataStore.
-func New(storage store.Store) DataStore {
-	return &datastoreImpl{
-		storage: storage,
+func New(storage store.Store, indexer index.Indexer, searcher search.Searcher) DataStore {
+	ds := &datastoreImpl{
+		storage:           storage,
+		indexer:           indexer,
+		formattedSearcher: searcher,
 	}
+
+	ctx := sac.WithGlobalAccessScopeChecker(context.Background(),
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(genSt.Access_READ_ACCESS, genSt.Access_READ_WRITE_ACCESS),
+			sac.ResourceScopeKeys(resources.ImageIntegration)))
+
+	if err := ds.buildIndex(ctx); err != nil {
+		log.Error(err.Error())
+		return nil
+	}
+	return ds
+}
+
+// NewForTestOnly returns an instance of DataStore only for tests.
+func NewForTestOnly(t *testing.T, storage store.Store, indexer index.Indexer, searcher search.Searcher) DataStore {
```

Comment:
Instead of this, in the test, you can just do
```
ds := &datastoreImpl{
	storage:           storage,
	indexer:           indexer,
	formattedSearcher: searcher,
}
```

directly. Unless this is used in a different package as well?